### DISCUSSION TITLE
The error underline looks bad

### DISCUSSION THREAD

**Discussion by widersky** [zed-industries/zed#26198] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Just look at any underline decoration 😅

My `settings.json`:

```json
{
  "show_copilot_suggestions": true,
  "terminal": {
    "dock": "bottom"
  },
  "project_panel": {
    "dock": "left",
    "folder_icons": false
  },
  "buffer_font_family": "Fira Code",
  "buffer_font_size": 12,
  "ui_font_family": "Inter",
  "ui_font_size": 15,
  "theme": "One Light",
  "autosave": "off",
  "format_on_save": "off",
  "buffer_line_height": "comfortable",
  "tab_size": 2,
  "confirm_quit": true,
  "show_whitespaces": "all",
  "wrap_guides": [100],
  "features": {
    "copilot": true
  },
  "tabs": {
    "close_position": "left",
    "git_status": true
  },
  "assistant": {
  }
}
```

### Environment

Zed: v0.121.3 (Zed Preview)
OS: macOS 14.3.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![image](https://github.com/zed-industries/zed/assets/45122321/8d9b71f8-0927-4c5b-94ee-7164c31f19a9)

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by fdemb** [comment]
The issue seems to occur only when the active display is LoDPI.
MacBook's 2x scaled HiDPI display:
<img width="262" alt="image" src="https://github.com/zed-industries/zed/assets/69842744/915e6bca-5a75-4c6d-9b94-cdcbd98a48ed">
External 1x 1440p LoDPI display:
![image](https://github.com/zed-industries/zed/assets/69842744/bbeb7b72-788d-45ed-aa70-943e7c484881)

**Comment by widersky** [comment]
![image](https://github.com/zed-industries/zed/assets/45122321/902ebc57-5b80-41c1-9b76-7186a0680e05)

This is how it's looking when issue occurs on single character - is almost invisible.

**Comment by doctopus** [comment]
Is there a setting to disable the error underlines completely?

**Comment by KraXen72** [comment]
this also happens for me on fedora 40, kde plasma 6, wayland on a 1440p 100% scaled display.

**Comment by forGGe** [comment]
Would be really nice if the style of diagnostic errors/warnings was customizable. I believe #365 / #366  made the style hardcoded as wavy, and I can't find any settings related to changing that.

**Comment by Braymatter** [comment]
This also happens for me. 

I would love to have the error text actually inlay under the line. I hate having to have either a full editor pane dedicated to errors, or hover my mouse over the line

**Comment by superdoer** [comment]
Can we please have a setting to disable the red underlines while typing? It is so distracting and frustrating. It's fine for diagnostics and spellchecks to happen with 2 seconds delay after I pause typing. I hate this sort of design. If nothing, at least change the color to yellow or something. Red is just pure bad.

**Comment by Ziehnert** [comment]
Would it be possible to make it configurable instead of keeping it hardcoded? I personally quite liked the straight lines before, and the style of them changing when working with different setups is really quite irritating

**Comment by milkowski** [comment]
Exactly, there should be an option to have preferred underline scale and possibly style. Currently in my setup the predefined scale is broken.
![obraz](https://github.com/user-attachments/assets/311306e1-d3a0-48f1-8627-da457a919505)

**Comment by jf908** [comment]
It seems like the scale of the squiggles scales with DPI so it only renders correctly at 2x DPI. It looks particularly bad at 1x because you can only see the top half of the squiggle so you just end up with several disconnected humps.
